File: grind75/Jan14/621.go
func leastInterval(tasks []byte, n int) int {
    m := [26]int{}
    maxt := -1
    cnt := -1
    for _, c := range tasks {
        idx := int(c-'A')
        m[idx]++
        if m[idx] > maxt {
            maxt = m[idx]
            cnt = 1
        } else if m[idx] == maxt {
            cnt++
        }
        
    }
    
    return max(len(tasks), (maxt-1)*(n+1)+cnt)
}

File: grind75/Jan14/621.py
from collections import defaultdict
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq = defaultdict(int)
        max_count = 0
        most_task = -1
        for task in tasks:
            freq[task] += 1
            if freq[task] > most_task:
                most_task = freq[task]
                max_count = 1
            elif freq[task] == most_task:
                max_count += 1
        
        return max(len(tasks), (n + 1) * (most_task - 1) + 
max_count)
